app/react/Attachments/components/UploadSupportingFile.tsx

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable react/prop-types */
+
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+
+import { Translate } from 'app/I18N';
+import { Icon } from 'UI';
+
+import { uploadAttachment } from '../actions/actions';
+import AttachmentsModal from './AttachmentsModal';
+
+interface UploadSupportingFileProps {
+  entityId: string;
+  entitySharedId: string;
+  storeKey: string;
+  progress?: any;
+}
+
+const UploadSupportingFile: React.FC<UploadSupportingFileProps> = props => {
+  const { entitySharedId, entityId, storeKey, progress } = props;
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const openModal = () => setModalOpen(true);
+  const closeModal = () => setModalOpen(false);
```

Comment:
Is it possible to have only one function with a boolean argument?

Reply:
The only way I know for doing that would be for example: 
```
 <button
  type="button"
  onClick={setModalOpen.bind(this, true)}
  className="upload-button btn btn-success attachments-modal-trigger"
>
```

But that generates a new onClick function every time so since is a different function react re-renders the component every time.

Not sure if there is a different approach I don't know.